src/smc-webapp/smc_chat.tsx

```diff
@@ -740,296 +835,550 @@ const ChatRoom = rclass(function({ name }) {
+              display: "flex",
+              flexDirection: "column"
             }}
           >
-            <div style={{ display: "flex", height: "6em" }}>
-              <FormControl
-                style={{ width: "85%", height: "100%" }}
-                autoFocus={false}
-                componentClass="textarea"
-                ref="input"
-                onKeyDown={e => {
-                  mark_as_read();
-                  return this.on_keydown(e);
-                }}
-                value={this.props.input}
-                placeholder={"Type a message..."}
-                onChange={e => this.props.actions.set_input(e.target.value)}
-              />
+            {!IS_MOBILE ? (
               <Button
-                style={{ width: "15%", height: "100%" }}
-                onClick={this.button_send_chat}
+                onClick={this.button_on_click}
                 disabled={this.props.input === ""}
-                bsStyle="success"
+                bsStyle="info"
+                style={{ height: "50%", width: "100%" }}
               >
-                <Icon name="chevron-circle-right" />
+                Preview
               </Button>
-            </div>
-            <div style={{ color: "#888", padding: "5px" }}>
-              Shift+enter to send. Double click to edit. Use{" "}
-              <a
-                href="https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/"
-                target="_blank"
-              >
-                Markdown
-              </a>{" "}
-              and{" "}
-              <a
-                href="https://en.wikibooks.org/wiki/LaTeX/Mathematics"
-                target="_blank"
-              >
-                LaTeX
-              </a>.
-            </div>
-          </div>
-        </div>
-      );
-    }
-  };
-});
-
-// Component for use via React
-exports.SideChat = function({ path, redux, project_id }) {
-  const name = redux_name(project_id, path);
-  const file_use_id = require("smc-util/schema").client_db.sha1(
-    project_id,
-    path
-  );
-  const actions = redux.getActions(name);
-  return (
-    <ChatRoom
-      redux={redux}
-      actions={redux.getActions(name)}
-      name={name}
-      project_id={project_id}
-      path={path}
-      file_use_id={file_use_id}
-    />
-  );
-};
-
-// Fitting the side chat into non-react parts of SMC:
-
-const render = function(redux, project_id, path) {
-  const name = redux_name(project_id, path);
-  const file_use_id = require("smc-util/schema").client_db.sha1(
-    project_id,
-    path
-  );
-  const actions = redux.getActions(name);
-  return (
-    <ChatRoom
-      redux={redux}
-      actions={actions}
-      name={name}
-      project_id={project_id}
-      path={path}
-      file_use_id={file_use_id}
-    />
-  );
-};
-
-// Render the given chatroom, and return the name of the redux actions/store
-exports.render = function(project_id, path, dom_node, redux) {
```

Comment:
Is this no longer used anywhere in our code base?

Reply:
Those were only used before **you** did the top-down react rewrite....

Reply:
I think some non-react editors opened side chat with this. However I remember now that you changed how side chat opened.